apple/Sources/Sargon/Protocols/EntityProtocol.swift

```diff
@@ -26,26 +28,61 @@ public protocol EntityProtocol: BaseEntityProtocol, CustomStringConvertible, Ide
 #endif // DEBUG
 }
 
-extension EntityProtocol {
+extension EntityBaseProtocol {
```

Comment:
Why isn't the logic in this extension part of the Rust code?

Reply:
because UniFFI does not support Traits -> Protocols on Structs. And I have not had time to impl some of the functions in Rust land yet.

The work of this PR was done over the weekend :D and it is actually not part of Stage 1, but rather Stage 2...

So YES the plan is ofc to always have ALL logic implemented in Rust, like the impl of these Swift code blocks

```
	public var virtualHierarchicalDeterministicFactorInstances: Set<HierarchicalDeterministicFactorInstance> {
	public var hasAuthenticationSigningKey: Bool
	public var deviceFactorSourceID: FactorSourceIDFromHash? {
```

With every PR into Sargon more and more code is migrated over.